src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowTests.cs

```diff
@@ -2034,6 +2034,386 @@ static bool Set(out int x)
+    }
+
+    bool M(object obj) { return true; }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (9,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(9, 15),
+                // (17,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(17, 15),
+                // (25,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 3
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(25, 15));
+        }
+
+        [Fact]
+        public void CondAccess_NullCoalescing_09()
+        {
+            var source = @"
+class C
+{
+
+    void M1(C c1)
+    {
+        int x, y;
+        _ = (c1?[x = y = 0]) ?? false
+            ? x.ToString()
+            : y.ToString(); // 1
+    }
+
+    void M2(C c1)
+    {
+        int x, y;
+        _ = (c1?[x = y = 0]) ?? true
+            ? x.ToString() // 2
+            : y.ToString();
+    }
+
+    public bool this[int x] => false;
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (10,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(10, 15),
+                // (17,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(17, 15));
+        }
+
+        [Fact]
+        public void CondAccess_NullCoalescing_10()
+        {
+            var source = @"
+class C
+{
+
+    void M1(C c1)
+    {
+        int x, y;
+        _ = (bool?)null ?? (c1?.M(x = y = 0) ?? false)
+            ? x.ToString()
+            : y.ToString(); // 1
+    }
+
+    void M2(C c1)
+    {
+        int x, y;
+        _ = (bool?)null ?? (c1?.M(x = y = 0) ?? true)
+            ? x.ToString() // 2
+            : y.ToString();
+    }
+
+    bool M(object obj) { return true; }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (10,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(10, 15),
+                // (17,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(17, 15));
+        }
+
+        [Fact]
+        public void NullCoalescing_ConditionalLeft()
```

Comment:
Consider some variations of this test that actually allow accessing an output.

Reply:
Tried to come up with some meaningful variations in here, but it's limited by the fact that conditional state is unsplit after visiting the LHS.